pinot-core/src/main/java/org/apache/pinot/core/indexsegment/mutable/MutableSegmentImpl.java

```diff
@@ -208,7 +208,7 @@ public long getLatestIngestionTimestamp() {
         // each forward index entry will contain a 4 byte dictionary ID
         forwardIndexColumnSize = FieldSpec.DataType.INT.size();
         int dictionaryColumnSize;
-        if (dataType == FieldSpec.DataType.STRING) {
+        if (!isFixedWidthColumn) {
```

Comment:
better switch the if/else.

also, good to add BYTES column in the schema for realtime test (can be another PR)